test/models/review_test.rb

```diff
@@ -929,24 +929,36 @@ def finding.can_be_destroyed?; true; end
     Current.user = users :supervisor
     cois         = control_objective_items :management_dependency_item_editable
 
-    cois.work_papers.create!(
-        code: 'PTOC 300',
-        name: 'New recode',
-        description: 'New workpaper description'
+    add_wp = cois.work_papers.create!(
+              code: 'PTOC 300',
+              name: 'New recode',
+              description: 'New workpaper description',
+              file_model_attributes: {
+                file: Rack::Test::UploadedFile.new(TEST_FILE_FULL_PATH)
+              }
     )
 
-    work_papers        = @review.work_papers.map &:code
-    recode_work_papers = @review.recode_work_papers.map &:code
-    codes              = {}
+    work_papers           = @review.work_papers.map &:code
+    recode_work_papers    = @review.recode_work_papers.map
+    codes                 = {}
+    work_papers_new_codes = recode_work_papers.map &:code
 
-    assert_not_equal work_papers, recode_work_papers
+    assert_not_equal work_papers, work_papers_new_codes
 
     recode_work_papers.sort.each do |wp|
-      prefix, code_number = wp.split
+      prefix, code_number = wp.code.split
       codes[prefix]     ||= 1
       test_code           = "#{prefix} #{'%.3d' % codes[prefix]}"
+      new_code            = test_code.sub(/\s/, '_')
 
-      assert_equal wp, test_code
+      assert_equal wp.code, test_code
+
+      if wp.file_model
+        code_file = wp.file_model.file_file_name.split('-')
+
+        assert wp.file_model.file_file_name.start_with? new_code
+        assert code_file.exclude? add_wp.code
```

Comment:
Esto debería ser igual que el de arriba:
```suggestion
        assert wp.file_model.file_file_name.exclude? add_wp.code
```

Porque deberías preguntar en todo el nombre del archivo. Le hago el merge porque necesito el fix para otro cliente, pero subí este arreglo a master.